### PR TITLE
allow hive engine owner to create and update pool for utility token

### PR DESCRIPTION
**Pull request by eonwarped**

### CODE REVIEW

File: contracts/mining.js [hunk 1]
```diff
@@ -223,7 +223,7 @@ actions.updatePool = async (payload) => {
   // eslint-disable-next-line no-template-curly-in-string
   const utilityTokenBalance = await api.db.findOneInTable('tokens', 'balances', { account: api.sender, symbol: "'${CONSTANTS.UTILITY_TOKEN_SYMBOL}$'" });
 
-  const authorized = api.BigNumber(poolUpdateFee).lte(0)
+  const authorized = api.BigNumber(poolUpdateFee).lte(0) || api.sender === api.owner
```
Comment (bt-cryptomancer) [inline contracts/mining.js:226]: Here you are skipping the balance check for the fee if api.owner is calling the contract action, but fee is still deducted further down.  Maybe api.owner should be excluded from paying the fee as it's a special case?

File: contracts/mining.js [hunk 2]
```diff
@@ -291,7 +292,7 @@ actions.createPool = async (payload) => {
   // eslint-disable-next-line no-template-curly-in-string
   const utilityTokenBalance = await api.db.findOneInTable('tokens', 'balances', { account: api.sender, symbol: "'${CONSTANTS.UTILITY_TOKEN_SYMBOL}$'" });
 
-  const authorizedCreation = api.BigNumber(poolCreationFee).lte(0)
+  const authorizedCreation = api.BigNumber(poolCreationFee).lte(0) || api.sender === api.owner
```
Comment (bt-cryptomancer) [inline contracts/mining.js:295]: Here you are skipping the balance check for the fee if api.owner is calling the contract action, but fee is still deducted further down.  Maybe api.owner should be excluded from paying the fee as it's a special case?

**Review by bt-cryptomancer — COMMENTED** [review]
Mostly looks good, just want clarification regarding fee handling for api.owner.

File: contracts/mining.js (hunk 1 above)
Comment (eonwarped) [reply contracts/mining.js:226]: Fixed

**Review by bt-cryptomancer — APPROVED** [review]
changes look good, thanks

**Pull request merged**